src/OrchardCore.Modules/OrchardCore.Search.Elastic/Services/ElasticIndexManager.cs

```diff
@@ -0,0 +1,246 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+using Microsoft.Extensions.Logging;
+
+using Nest;
+
+using OrchardCore.Contents.Indexing;
+using OrchardCore.Indexing;
+using OrchardCore.Modules;
+using OrchardCore.Search.Elastic.Services;
+
+namespace OrchardCore.Search.Elastic
+{
+    /// <summary>
+    /// Provides methods to manage physical Lucene indices.
+    /// This class is provided as a singleton to that the index searcher can be reused across requests.
+    /// </summary>
+    public class ElasticIndexManager : IDisposable
+    {
+        private readonly IElasticClient _elasticClient;
+
+       
+        private readonly IClock _clock;
+        private readonly ILogger _logger;
+        private bool _disposing;
+        private ConcurrentDictionary<string, DateTime> _timestamps = new ConcurrentDictionary<string, DateTime>(StringComparer.OrdinalIgnoreCase);
+        private readonly ElasticAnalyzerManager _elasticAnalyzerManager;
+        private readonly ElasticIndexSettingsService _elasticIndexSettingsService;
+
+        public ElasticIndexManager(
+            IClock clock,
+            ILogger<ElasticIndexManager> logger,
+            ElasticAnalyzerManager elasticAnalyzerManager,
+            ElasticIndexSettingsService elasticIndexSettingsService,
+            IElasticClient elasticClient
+            )
+        {
+            
+            _clock = clock;
+            _logger = logger;
+            _elasticAnalyzerManager = elasticAnalyzerManager;
+            _elasticIndexSettingsService = elasticIndexSettingsService;
+            _elasticClient = elasticClient;
+        }
+
+        public async Task<bool> CreateIndexAsync(string indexName)
+        {
+            //Get Index name scoped by ShellName
+            if (!await Exists(indexName))
+            {
+                var response = await _elasticClient.Indices.CreateAsync(indexName);
+                return response.Acknowledged;
```

Comment:
This could use some Exception catching here as it currently silently fails when connecting to OpenSearch for example.